spacy/kb/kb_in_memory.pyx

```diff
@@ -238,7 +246,7 @@ cdef class InMemoryLookupKB(KnowledgeBase):
         alias_index = <int64_t>self._alias_index.get(alias_hash)
         alias_entry = self._aliases_table[alias_index]
 
-        return [Candidate(kb=self,
+        return [Candidate(retrieve_string_from_hash=self.vocab.strings.__getitem__,
```

Comment:
Mentioned above, but: this is a breaking change (although I doubt this would be noticed) and I don't particularly like this solution. It'd be cleaner to just pass the alias, since I don't think it'd occur a significant memory overhead to have `Candidate` instances store their mentions/aliases instead of looking them up. This would be even more (?) breaking, as it would eliminate the need for hashes. 

As mentioned I don't think many devs are writing their own KBs (deriving from the published spaCy KB) anyway, so the more breaking version would be preferrable to me.